### DISCUSSION TITLE
Formatting Values in the Dashboard

### DISCUSSION THREAD

**Discussion by CrossBound** [datalust/seq-tickets#1082] [Ideas]
It appears that each dashboard widget only works with numeric values... It would be nice if we could add text fields that would be displayed in the "value" type widget. At the very least if we could specify a format string for the value. In my particular case I'm trying to display a monetary field using LAST(ROUND(amount, 0)) AS Amount but it doesn't allow me to format it with the USD ($) currency symbol.

**Comment by larenelg** [comment]
Love this idea! Will chat about possibilities around adding units with the team. Thanks 👍

**Comment by nblumhardt** [comment]
Hi @CrossBound - does:

```sql
concat('$', ToJson(Round(last(amount), 0)))
```

do what you're after?

(`ToJson()` is a sneaky way to stringify numbers, in the absence of `ToString()` --- which we probably should add 😅 )

**Comment by CrossBound** [comment]
That sort-of works. It does append the currency symbol ($) but then I lose my comma separated groupings which makes it harder to read.

Before
 
![image](https://user-images.githubusercontent.com/11417093/98959054-86e2c600-24c8-11eb-81a1-3d5f0bae576e.png)

After

![image](https://user-images.githubusercontent.com/11417093/98959076-8d713d80-24c8-11eb-8bb2-549daa65629b.png)

**Comment by CrossBound** [comment]
So it looks like I can achieve my goal by doing some trickery like `CONCAT('$',CONCAT(SUBSTRING(TOJSON(LAST((Amount % 1000000000) / 1000000)), 0, INDEXOF(TOJSON(LAST((Amount % 1000000000) / 1000000)), '.')),CONCAT(CONCAT(',',SUBSTRING(TOJSON(LAST((Amount % 1000000) / 1000)), 0, INDEXOF(TOJSON(LAST((Amount % 1000000) / 1000)), '.'))), CONCAT(',',TOJSON(LAST(ROUND(Amount % 1000, 0)))))))`

But that seems pretty outlandish to have to resort to such measures. It would be much easier if I could just query `LAST(Amount)` and provide a format string to the display.

**Comment by nblumhardt** [comment]
Ah, I see! Some pretty big numbers you're dealing with, there :-)

We'll need to give it a bit more thought 👍

**Comment by CrossBound** [comment]
Curious if there are any updates on this?

**Reply by nblumhardt** [reply]
Thanks for the nudge, David. Unfortunately, no; we have the added challenge now that not all code responsible for evaluating expressions is using .NET, so we'll additionally need to decide on a common formatting syntax. Good for us to think about...
